plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryQueryPageSource.java

```diff
@@ -102,7 +106,7 @@ public BigQueryQueryPageSource(
                 client.getProjectId(),
                 ImmutableList.copyOf(columnHandles),
                 filter);
-        this.tableResult = client.executeQuery(session, sql);
+        this.tableResult = CompletableFuture.supplyAsync(() -> client.executeQuery(session, sql), executor);
```

Comment:
Record the time taken to  executeQuery in the future and report it in `getReadTimeNanos`. Without that it will be hard to understand why the scan took a certain amount of time.

Reply:
Done!